### PR TITLE
feat: Adds `max_healthy_percentage` and `skip_matching` to autoscaling group

### PR DESCRIPTION
**Pull request by easterbonbon**
## Description

Adds `max_healthy_percentage` and `skip_matching` to `aws_autoscaling_group` `instance_refresh.preferences`. 

Small aside: also updated the examples to look for an Amazon Linux 2 AMI rather than an Amazon Linux AMI which is EOL.

## Motivation and Context

These two settings were added in the AWS Terraform provider and currently there is no way to specify them using this module.

`max_healthy_percentage` was added in v5.32.0 ([see changelog](https://github.com/hashicorp/terraform-provider-aws/blob/a1bb50593b1a1bbcd73b997ead17dd77a2d32c0b/CHANGELOG.md?plain=1#L268))
and
`skip_matching` was added in v4.13.0 ([see changelog](https://github.com/hashicorp/terraform-provider-aws/blob/23f68e440e1456cbc19fbdeec974a1580f104c25/CHANGELOG.md?plain=1#L2224))

## Breaking Changes

No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.26"
+      version = ">= 5.32"
```
Comment (bryantbiggs) [inline versions.tf:7]: we need to change this throughout - aligning all of the examples as well

Comment (easterbonbon) [reply versions.tf:7]: Thanks! I've updated the required version in the `examples` and updated the docs. Let me know if there anything else I've missed :)

**Review by bryantbiggs — APPROVED** [review]
awesome, thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 7.4.0 :tada:

**Pull request merged**